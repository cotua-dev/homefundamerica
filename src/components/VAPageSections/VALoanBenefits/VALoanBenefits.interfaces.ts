export interface Benefit {
    key: number;
    benefit: string;
}
