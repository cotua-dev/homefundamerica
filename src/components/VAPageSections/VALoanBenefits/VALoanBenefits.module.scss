@import '../../../styles/variables';

section#vaLoanBenefitsSection {
    height: 50vh;
    display: flex;
    align-items: center;
    justify-content: center;

    @media screen and (max-width: 950px) {
        height: 100%;
        padding: 40px 10px;
    }

    div.contentWrapper {
        width: 600px;

        @media screen and (max-width: 767px) {
            width: 100%;
        }

        h2.loanBenefitsTitle {
            color: $primaryColor;
            margin: 0;
            font-size: 32px;
            text-align: center;

            @media screen and (max-width: 950px) {
                font-size: 24px;
            }
        }

        p.content {
            color: $primaryColor;
            font-size: 16px;

            @media screen and (max-width: 950px) {
                text-align: center;
                font-size: 14px;
            }
        }

        div.benefitsWrapper {
            display: flex;
            flex-direction: row;
            margin-top: 20px;

            @media screen and (max-width: 950px) {
                flex-direction: column;
            }

            div.benefits {
                div.benefit {
                    margin: 10px 0;

                    strong,
                    .icon {
                        font-size: 20px;

                        @media screen and (max-width: 950px) {
                            font-size: 14px;
                        }
                    }

                    strong {
                        color: $primaryColor;
                    }

                    .icon {
                        color: $secondaryColor;
                        margin-right: 10px;
                    }
                }
            }
        }

        strong.service {
            font-size: 32px;
            color: $primaryColor;
            text-align: center;
            display: flex;
            justify-content: center;
            margin-top: 20px;

            @media screen and (max-width: 950px) {
                font-size: 24px;
            }
        }
    }
}
