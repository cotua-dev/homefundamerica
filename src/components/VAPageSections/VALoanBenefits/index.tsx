import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck } from '@fortawesome/free-solid-svg-icons';
import * as styles from './VALoanBenefits.module.scss';
import { Benefit } from './VALoanBenefits.interfaces';

export const VALoanBenefits: FunctionComponent = (): ReactElement => {
    const benefitsSetOne: Benefit[] = [
        {
            key: 0,
            benefit: 'No Money Down',
        },
        {
            key: 1,
            benefit: 'No Mortgage Insurance',
        },
        {
            key: 2,
            benefit: 'Loan Amounts to $1.5 Million',
        },
        {
            key: 3,
            benefit: 'Credit Scores as Low as 500',
        },
        {
            key: 4,
            benefit: 'Competitive Rates',
        },
        {
            key: 5,
            benefit: 'Great Loans for First-Time Homebuyers',
        },
    ];

    return (
        <section id={styles.vaLoanBenefitsSection}>
            <div className={styles.contentWrapper}>
                <h2 className={styles.loanBenefitsTitle}>VA Loan Benefits</h2>
                <p className={styles.content}>
                    {`
                        VA Loans offer unique benefits to eligible veterans and
                        active duty servicemen and women who are looking for:
                    `}
                </p>
                <div className={styles.benefitsWrapper}>
                    <div className={styles.benefits}>
                        {benefitsSetOne.map((benefit: Benefit) => (
                            <div className={styles.benefit} key={benefit.key}>
                                <FontAwesomeIcon className={styles.icon} icon={faCheck}/>
                                <strong>{benefit.benefit}</strong>
                            </div>
                        ))}
                    </div>
                </div>
                <strong className={styles.service}>
                    {`Thank you for your service!`}
                </strong>
            </div>
        </section>
    );
}
