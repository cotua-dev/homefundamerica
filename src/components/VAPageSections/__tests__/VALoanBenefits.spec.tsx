import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { VALoanBenefits } from '../VALoanBenefits';

describe('VALoanBenefits Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<VALoanBenefits/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
