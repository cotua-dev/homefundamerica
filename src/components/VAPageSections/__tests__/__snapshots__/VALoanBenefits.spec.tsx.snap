// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VALoanBenefits Component renders 1`] = `
<section>
  <div>
    <h2>
      VA Loan Benefits
    </h2>
    <p>
      
                        VA Loans offer unique benefits to eligible veterans and
                        active duty servicemen and women who are looking for:
                    
    </p>
    <div>
      <div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            No Money Down
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            No Mortgage Insurance
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Loan Amounts to $1.5 Million
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Credit Scores as Low as 500
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Competitive Rates
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Great Loans for First-Time Homebuyers
          </strong>
        </div>
      </div>
    </div>
    <strong>
      Thank you for your service!
    </strong>
  </div>
</section>
`;
